framework/src/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -43,28 +45,75 @@ public static ContractsUtils getInstance(GenericAnnotatedTypeFactory<?, ?, ?, ?>
         return instance;
     }
 
-    /**
-     * Represents a pre- or postcondition that must be verified by {@code BaseTypeVisitor} or one of
-     * its subclasses. Automatically extracted from annotations with meta-annotations
-     * {@code @PreconditionAnnotation} or {@code @PostconditionAnnotation}, such as
-     * {@code @RequiresNonNull} and {@code @EnsuresNonNull}. Can also be generated, such as in
-     * {@code LockVisitor.generatePreconditionsBasedOnGuards}.
-     */
-    // When making changes, make also the appropriate changes
-    // to class ConditionalPostcondition.
-    public static class PreOrPostcondition {
+    public abstract static class Contract {
```

Comment:
Javadoc for the class, enum, and fields would be useful. It is not obvious what kind of annotation is contained in the `annotation` field and some javadoc would help (maybe also a more specific field name.).